### PR TITLE
keep all operands in the same namespace

### PR DESCRIPTION
**Pull request by phoracek**
Keeping operands in different namespaces is not supportable on
OpenShift with OLM. Let's keep all we deploy in a single namespace.
By default, the namespace of the operator itself will be used.

### CODE REVIEW

File: pkg/components/components.go [hunk 1]
```diff
@@ -146,6 +146,14 @@ func GetDeployment(version string, operatorVersion string, namespace string, rep
 										},
 									},
 								},
+								{
+									Name: "OPERAND_NAMESPACE",
```
Comment (SchSeba) [inline pkg/components/components.go]: why we need this?

can't we just use the `OPERATOR_NAMESPACE`

File: pkg/controller/networkaddonsconfig/networkaddonsconfig_controller.go [hunk 2]
```diff
@@ -310,11 +312,13 @@ func (r *ReconcileNetworkAddonsConfig) renderObjects(networkAddonsConfig *opv1al
 // are removed when NetworkAddonsConfig config is
 func (r *ReconcileNetworkAddonsConfig) applyObjects(networkAddonsConfig *opv1alpha1.NetworkAddonsConfig, objs []*unstructured.Unstructured) error {
 	for _, obj := range objs {
-		// Mark the object to be GC'd if the owner is deleted
-		if err := controllerutil.SetControllerReference(networkAddonsConfig, obj, r.scheme); err != nil {
-			log.Printf("could not set reference for (%s) %s/%s: %v", obj.GroupVersionKind(), obj.GetNamespace(), obj.GetName(), err)
-			err = errors.Wrapf(err, "could not set reference for (%s) %s/%s", obj.GroupVersionKind(), obj.GetNamespace(), obj.GetName())
-			return err
+		// Mark the object to be GC'd if the owner is deleted. Don't set owner reference on namespaces if they are used by the operator itself
```
Comment (SchSeba) [inline pkg/controller/networkaddonsconfig/networkaddonsconfig_controller.go]: nice comment!

File: pkg/components/components.go (hunk 1 above)
Comment (phoracek) [reply pkg/components/components.go]: We could, but what if we would need to use two different namespaces then? If we keep it split like this, all our components will work with the assumption those namespaces are not always identical. Maybe I'm too paranoid though.

File: pkg/controller/networkaddonsconfig/networkaddonsconfig_controller.go (hunk 2 above)
Comment (phoracek) [reply pkg/controller/networkaddonsconfig/networkaddonsconfig_controller.go]: :cat:

**Pull request merged**